src/utilities/validation.ts

```diff
@@ -80,3 +79,38 @@ export function validateDateWithInterval(
   }
   return { msg: "", err: false };
 }
+
+export function validateNationalIDCardNumber(nationalIDCardNumber: string): validation {
+  if (nationalIDCardNumber.length == 0) {
+    return { msg: "ช่องนี้ไม่สามารถเว้นว่างได้", err: true };
+  }
+
+  if (
+    nationalIDCardNumber.length != CHAR_LIMIT.MAX_NATIONAL_ID_CARD_NUMBER ||
+    !RegExp(/\d{13}/).test(nationalIDCardNumber)
+  ) {
+    return { msg: "รูปแบบเลขบัตรประจำตัวประชาชนไม่ถูกต้อง", err: true };
+  }
+
+  if (!checkLastDigit(nationalIDCardNumber)) {
+    return { msg: "รูปแบบเลขบัตรประจำตัวประชาชนไม่ถูกต้อง", err: true };
+  }
```

Comment:
Maybe these should return different messages or you could merge them into one if since they return the same thing.

Reply:
I will merge them to same thing.